### PR TITLE
Set up a "Hello world" project as one app with Webpack

### PR DESCRIPTION
**Pull request by kambereBr**
### Project requirements

  ✅  Implement a connection between a Ruby on Rails back-end and React front-end.
  ✅ Understand the pros and cons of different approaches of connecting Ruby on Rails back-end with React front-end.
  ✅ Created a new Rails app called 'hello-rails-react'.
  ✅ Verified that the `jsbundling-rails` gem is added to your Gemfile.
  ✅ Created a `babel.config.js` in the root folder of your project as described in the previous lesson's tutorial.
  ✅ Updated `webpack.config.js` to include the babel loader as described in the previous lesson's tutorial.
  ✅ Made sure that your project has a Postgres database set up. 
  ✅ Created a table for storing your messages and create 5 different greetings. 
  ✅ Created an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
  ✅Created a static view that will be the root of your app.

---------

 ✅Created the `App` component with react-router and render it in your static view.
 ✅ Created the `Greeting` component that will display a greeting. Set it up as a route in your App component.
 ✅ Created a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
 ✅ Displayed the random greeting in your `Greeting` component.

### CODE REVIEW

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @kambereBr, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Use jsbundling-rails ✔️ 
- Use webpack ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please 🙏🏽 check your gems to be able to run your app. I was trying several times to fix the error without success. 👇🏽 

```rails
franclobo@DESKTOP-E09US2D:~/microverse/hello-rails-react-app$ ./bin/dev
21:23:25 web.1  | started with pid 32156
21:23:25 js.1   | started with pid 32157
21:23:25 js.1   | yarn run v1.22.19
21:23:25 js.1   | $ webpack --config webpack.config.js --watch
21:23:26 web.1  | C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/definition.rb:540:in `materialize': Could not find rails-7.1.2, sprockets-rails-3.4.2, puma-6.4.0, jsbundling-rails-1.2.1, turbo-rails-1.5.0, stimulus-rails-1.3.0, jbuilder-2.11.5, tzinfo-data-1.2023.3, bootsnap-1.17.0, debug-1.8.0, web-console-4.2.1, capybara-3.39.2, selenium-webdriver-4.15.0, actioncable-7.1.2, actionmailbox-7.1.2, actionmailer-7.1.2, actionpack-7.1.2, actiontext-7.1.2, actionview-7.1.2, activejob-7.1.2, activemodel-7.1.2, activerecord-7.1.2, activestorage-7.1.2, activesupport-7.1.2, railties-7.1.2, sprockets-4.2.1, nio4r-2.6.1, msgpack-1.7.2, irb-1.9.1, reline-0.4.1, bindex-0.8.1, addressable-2.8.5, mini_mime-1.1.5, rack-3.0.8, xpath-3.2.0, rubyzip-2.3.2, websocket-1.2.10, websocket-driver-0.7.6, zeitwerk-2.6.12, net-imap-0.4.7, net-smtp-0.4.0, rails-dom-testing-2.2.0, rack-session-2.0.0, globalid-1.2.1, timeout-0.4.1, base64-0.2.0, bigdecimal-3.1.4, connection_pool-2.4.1, drb-2.2.0, minitest-5.20.0, mutex_m-0.2.0, rackup-2.1.0, rake-13.1.0, thor-1.3.0, rdoc-6.6.0, public_suffix-5.0.4, net-protocol-0.2.2, loofah-2.22.0, webrick-1.8.1, psych-5.1.1.1, stringio-3.1.0 in locally installed gems (Bundler::GemNotFound)
21:23:26 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/definition.rb:200:in `specs'
21:23:26 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/definition.rb:266:in `specs_for'
21:23:26 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/runtime.rb:18:in `setup'
21:23:26 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler.rb:162:in `setup'
21:23:26 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/setup.rb:23:in `block in <top (required)>'
21:23:26 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/ui/shell.rb:159:in `with_level'
21:23:26 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/ui/shell.rb:111:in `silence'
21:23:26 web.1  |       from C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/bundler/setup.rb:23:in `<top (required)>'
21:23:26 web.1  |       from <internal:C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
21:23:26 web.1  |       from <internal:C:/Ruby32-x64/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
21:23:26 web.1  |       from //wsl.localhost/Ubuntu/home/franclobo/microverse/hello-rails-react-app/config/boot.rb:3:in `<top (required)>'
21:23:26 web.1  |       from bin/rails:3:in `require_relative'
21:23:26 web.1  |       from bin/rails:3:in `<main>'
21:23:27 web.1  | exited with code 1
21:23:27 system | sending SIGTERM to all processes
21:23:27 js.1   | exited with code 1
```

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kambereBr** [comment]
Dear @franclobo, thanks for your review 👏 
Did you try to compile the app first before launching it? Because actually, the app works fine on my side.

Thanks !

demo video:


[rails-react-video-demo_edit_0.webm](https://github.com/kambereBr/hello-rails-react-app/assets/29512462/6f3b17f3-ac5d-4e29-a7e0-7b272d13d22c)

File: config/routes.rb [hunk 1]
```diff
@@ -1,10 +1,11 @@
 Rails.application.routes.draw do
+  get 'greetings/api'
   root 'root#index'
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Reveal health status on /up that returns 200 if the app boots with no exceptions, otherwise 500.
   # Can be used by load balancers and uptime monitors to verify that the app is live.
-  get "up" => "rails/health#show", as: :rails_health_check
+  get 'up' => 'rails/health#show', as: :rails_health_check
```
Comment (FrederickMih) [inline config/routes.rb]: - [Optional] Let's kindly keep the codebase clean. These are unnecessary routes.

File: app/javascript/src/redux/greetingsSlice.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export const getGreetings = createAsyncThunk(
+  'greetings/getGreetings',
+  async () => {
+    try {
+      const response = await axios.get('/greetings/api');
```
Comment (FrederickMih) [inline app/javascript/src/redux/greetingsSlice.js:8]: - [Optional] URL from the backend doesn't look like an API: It is a good practice to prefix `api/v1` . This is important especially if we have plans to update our api in the future and still maintain the previous version :+1:

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: :raising_hand: @kambereBr, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 a

:tada: :champagne:  :tada: :champagne: 

## Highlights :green_circle: 

- PR is professional :+1: 
- Two apps connect together to display random greetings :+1: 

## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (kambereBr) [reply config/routes.rb]: Dear @FrederickMih, thanks for your review 👏

**Pull request merged**